evaluate/task/task-code-repair.go

```diff
@@ -0,0 +1,157 @@
+package task
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	pkgerrors "github.com/pkg/errors"
+	"github.com/symflower/eval-dev-quality/evaluate/metrics"
+	"github.com/symflower/eval-dev-quality/language"
+	"github.com/symflower/eval-dev-quality/log"
+	"github.com/symflower/eval-dev-quality/model"
+	evaltask "github.com/symflower/eval-dev-quality/task"
+)
+
+// TaskCodeRepair holds the code repair task.
+type TaskCodeRepair struct {
+	// ResultPath holds the directory path where results should be written to.
+	ResultPath string
+
+	// Language holds the language for which the task should be evaluated.
+	Language language.Language
+	// Model holds the model which the task should be evaluated.
+	Model model.Model
+
+	// Logger holds the logger for this tasks.
+	Logger *log.Logger
+}
+
+// TaskCodeRepairArguments holds extra arguments to be used in a query prompt.
+type TaskCodeRepairArguments struct {
+	// Mistakes holds the list of compilation errors for a repository.
+	Mistakes []string
+}
+
+var _ evaltask.Task = (*TaskCodeRepair)(nil)
+
+// newCodeRepairTask returns a code repair task.
+func newCodeRepairTask(logger *log.Logger, resultPath string, model model.Model, language language.Language) (task evaltask.Task) {
+	return &TaskCodeRepair{
+		ResultPath: resultPath,
+		Language:   language,
+		Model:      model,
+		Logger:     logger,
+	}
+}
+
+// Identifier returns the code repair task identifier.
+func (t *TaskCodeRepair) Identifier() evaltask.Identifier {
+	return IdentifierCodeRepair
+}
+
+// Run performs source code repairing in a repository with compilation errors.
+func (t *TaskCodeRepair) Run(repository evaltask.Repository) (repositoryAssessment metrics.Assessments, problems []error, err error) {
+	log, logClose, err := log.WithFile(t.Logger, filepath.Join(t.ResultPath, string(t.Identifier()), model.CleanModelNameForFileSystem(t.Model.ID()), t.Language.ID(), repository.Name()+".log"))
+	if err != nil {
+		return nil, nil, err
+	}
+	defer logClose()
+
+	var innerRepositories []string
+	files, err := os.ReadDir(repository.DataPath())
+	if err != nil {
+		return nil, nil, pkgerrors.WithStack(err)
+	}
+	for _, file := range files {
+		if file.IsDir() && !strings.HasPrefix(file.Name(), ".") {
+			innerRepositories = append(innerRepositories, filepath.Join(repository.DataPath(), file.Name()))
+		}
+	}
+
+	repositoryAssessment = metrics.NewAssessments()
+	for _, innerRepository := range innerRepositories {
+		if err := repository.Reset(t.Logger); err != nil {
+			t.Logger.Panicf("ERROR: unable to reset temporary repository path: %s", err)
+		}
+
+		filePaths, err := t.Language.Files(log, innerRepository)
+		if err != nil {
+			return nil, problems, pkgerrors.WithStack(err)
+		}
+		if len(filePaths) != 2 {
+			problems = append(problems, pkgerrors.Errorf("the inner repository must only contain an implementation file and the corresponding test file, but found %#v", filePaths))
```

Comment:
I think we should use "task case" for the wording. I know inner "repository" feels tempting cause it is essentially a standalone repository... but to us it is a task case / example / candidate.

Reply:
Also please report the name of the repository + test case path so we see that during debugging!